autolab/Autolab
# Add course number

## Description
- Add field to CUD table to store course number
- Display course number where relevant (gradebook / gradesheet / manage students / course profile)
- Update `add user to course` form to support course number
- Update roster import to support course numbers
- Add "by course" statistics to statistics page

Misc
- Prettify `course average tweak` section of `add user to course`
- Left-align emails in gradebook, remove redundant CSV (the classes don't exist, they're all `.tip`)
- Increase default width of columns in gradebook
- Remove dead reference to `assessments_show` in `show.html.erb`
- Lint Form Builder

## Motivation and Context
Record course numbers for students, so that students from different courses but in the same autolab course can be distinguished.

## How Has This Been Tested?
- Successfully import course number via autolab format / CMU roster format (for a new student)
- Successfully import course number via autolab format / CMU roster format (for an existing student). Essentially, upload the same roster for the previous part, but with some data changed (such as course number). After saving the roster, the changes should be reflected.
- Ensure that import success / error tables show course number (when it exists)
- Ensure that gradebook / manage student tables show course number (when it exists)
- Ensure that `add user to course` form is able to correctly create user with course number
- Ensure that you can edit the course number of an existing user

**Import Autolab Format / CMU Roster Format**
Autolab Format: `M22,xho@foo.bar,Ho,Damian,SCS,CS,2025,L,15122,1,A`
CMU Roster Format: `Semester,Course,Section,Lecture,Mini,Last Name,Preferred/First Name,MI,Andrew ID,Email,College,Department,Major,Class,Graduation Semester,Units,Grade Option,QPA Scale,Mid-Semester Grade,Primary Advisor,Final Grade,Default Grade,Time Zone Code,Time Zone Description,Added By,Added On,Confirmed,Waitlist Position,Units Carried/Max Units,Waitlisted By,Waitlisted On,Dropped By,Dropped On,Roster As Of Date
M22,15122,A,1,N,Ho,Damian,,xho,xho@andrew.cmu.edu,SCS,CS,CS,3,F25,0,L,4+,,Test Advisor,,,EST,Eastern Standard Time (GMT-5:00),fake_andrewid,11 Jul 2022,Y,,,,,,,11 Jul 2022 12:00 AM
`
<img width="1018" alt="Screenshot 2022-11-10 at 21 41 06" src="https://user-images.githubusercontent.com/9074856/201251291-e4629726-9a6a-4eb9-8ad9-208a2a0c4575.png">

**Import error table**
Autolab Format: `M22,bad@foo.bar,,,SCS,CS,2025,L,15122,1,A`
<img width="994" alt="Screenshot 2022-11-10 at 21 41 43" src="https://user-images.githubusercontent.com/9074856/201251308-260a8c0f-9d26-468c-af58-83779bc5e5b4.png">

**Course number appears on gradebook / gradesheet / manage students / course profile**
(Note: gradebook's email cells are now left-aligned as was intended, cells are also wider)
<img width="987" alt="Screenshot 2022-11-10 at 21 41 55" src="https://user-images.githubusercontent.com/9074856/201251338-ade75090-1491-40e6-a3c9-629f2c6baaa1.png">
<img width="1010" alt="Screenshot 2022-11-10 at 21 42 04" src="https://user-images.githubusercontent.com/9074856/201251339-5d6faee9-357c-486d-9903-453407be1b9f.png">
<img width="1138" alt="Screenshot 2022-11-10 at 21 42 16" src="https://user-images.githubusercontent.com/9074856/201251341-b70f49b0-c0b9-43d4-8f70-f243fb236986.png">
<img width="1017" alt="Screenshot 2022-11-10 at 21 42 26" src="https://user-images.githubusercontent.com/9074856/201251343-f1ef348b-171f-4e52-b1ec-4e695bd6b82d.png">


**Add user to course form**
<img width="1028" alt="Screenshot 2022-11-06 at 15 16 04" src="https://user-images.githubusercontent.com/9074856/200193123-5318aa91-b548-484e-a312-e9b8032d5874.png">

**Statistics by course**
<img width="957" alt="Screenshot 2022-11-10 at 21 42 48" src="https://user-images.githubusercontent.com/9074856/201251357-917d143c-c749-4cae-b554-99eb0838bdbf.png">

## Types of changes

## Checklist:
